### PR TITLE
Coin drop animation + Support for different board sizes

### PR DESCRIPTION
**Pull request by amanpalariya**
- Coin drop animation has been added
- Different board sizes are supported
- Board pattern can be changed by changing the `board1_100px.png` image inside the images folder
- Some minor bug fixes (like coin drawing while undoing)

### CODE REVIEW

**Comment by bymayanksingh** [comment]
Great Work @amanpalariya but I guess you have done a lot of things which we may not need, currently reviewing your PR.

**Review by bymayanksingh — APPROVED** [review]

File: game_board.py [hunk 1]
```diff
@@ -153,4 +153,4 @@ def tie_move(self):
                 if self.board[r][c] != 0:
                     slots_filled += 1
 
-        return slots_filled == 42
+        return slots_filled == self.rows*self.cols
```
Comment (code-monk08) [inline game_board.py]: approved.

File: game_board.py [hunk 2]
```diff
@@ -28,7 +28,7 @@ def print_board(self):
         """
         print(flip(self.board, 0))
         print(" " + "---"*self.cols)
-        print(" " + str([1, 2, 3, 4, 5, 6, 7]))
+        print(" " + str([i+1 for i in range(self.cols)]))
```
Comment (code-monk08) [inline game_board.py]: approved

File: game_renderer.py [hunk 3]
```diff
@@ -177,13 +177,16 @@ def on_game_over(self, event: GameOver):
             self.label = self.myfont.render("GAME DRAW !!!!", 1, white)
             self.screen.blit(self.label, (40, 10))
 
-    def draw_board(self, board):
+    def draw_board(self, data):
         """
         Draws the game board to the screen.
         :param board: The game board.
         """
```
Comment (code-monk08) [inline game_renderer.py]: change this doc string according to new parameter data.

File: game_renderer.py [hunk 4]
```diff
@@ -177,13 +177,16 @@ def on_game_over(self, event: GameOver):
...
-        sq_size = 100
-        height = 700
+
+        board = data.game_board
+
+        sq_size = data.sq_size
+        height = data.height
```
Comment (code-monk08) [inline game_renderer.py]: approved

**Comment by bymayanksingh** [comment]
hey please [squash](https://stackoverflow.com/questions/35703556/what-does-it-mean-to-squash-commits-in-git) all of these commits/changes into one!

**Comment by bymayanksingh** [comment]
also I have so many concerns right now.. I will address them once you squash all your commits in one.

**Comment by amanpalariya** [comment]
I'm using git for the first time and don't know what I did just now 🙄. 
If there was a contiguous sequence of my commits, I squashed them into one. I hope that is what I was supposed to do.

**Comment by bymayanksingh** [comment]
> I'm using git for the first time and don't know what I did just now roll_eyes.
> If there was a contiguous sequence of my commits, I squashed them into one. I hope that is what I was supposed to do.

yes, you are supposed to do that only, in your fork I see that you made 13 commits, all I am asking is bring all those commits into **1** single commit and then raise a pull request, this PR is quite messy right now.. do one thing close this PR and raise a new one.. following everything I said in this comment. follow this [video](https://www.youtube.com/watch?v=2E23I9PzplM)

**Pull request closed without merging**